File: src/brasilburger/models/Quartier.java
package brasilburger.models;

public class Quartier {
    private int idQuartier;
    private String nom;
    private Zone zone;

    public Quartier() {
    }

    public Quartier(int idQuartier, String nom, Zone zone) {
        this.idQuartier = idQuartier;
        this.nom = nom;
        this.zone = zone;
    }

    public int getIdQuartier() {
        return idQuartier;
    }

    public void setIdQuartier(int idQuartier) {
        this.idQuartier = idQuartier;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public Zone getZone() {
        return zone;
    }

    public void setZone(Zone zone) {
        this.zone = zone;
    }
}

File: GestionRestau/GestionRestau/Models/Entities/Quartier.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BrasilBurger.Web.Models.Entities
{
    [Table("quartiers")]
    public class Quartier
    {
        [Key]
        [Column("id_quartier")]
        public int IdQuartier { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("nom")]
        public string Nom { get; set; } = string.Empty;

        [Column("id_zone")]
        public int IdZone { get; set; }

        // Navigation property
        [ForeignKey("IdZone")]
        public virtual Zone Zone { get; set; } = null!;
    }
}
